packages/ui/src/app/App.stories.tsx

```diff
@@ -606,10 +612,91 @@ export const BuyMembershipWithValidatorAccountHappyAndBindHappy: Story = {
+
+    await userEvent.click(getButtonByText(screen, 'Join Now'))
+
+    await step('Form', async () => {
+      const createButton = getButtonByText(modal, 'Create a Membership')
+
+      await step('Fill', async () => {
+        expect(createButton).toBeDisabled()
+        await fillMembershipFormWithTwoValidatorAcc(modal)
+        await waitFor(() => expect(createButton).toBeEnabled())
+      })
+
+      await userEvent.click(createButton)
+    })
+
+    await step('Create membership', async () => {
+      expect(modal.getByText('You intend to create a validator membership.'))
+      expect(modal.getByText('Creation fee:')?.nextSibling?.textContent).toBe('20')
+      expect(modal.getByText('Transaction fee:')?.nextSibling?.textContent).toBe('5')
+      expect(modal.getByRole('heading', { name: 'bob' }))
+
+      await userEvent.click(getButtonByText(modal, 'Create membership'))
+    })
+
+    await step('Add first validator account', async () => {
+      expect(await modal.findByText('You are intending to bond your validator account with your membership.'))
+      expect(modal.getByText('Transaction fee:')?.nextSibling?.textContent).toBe('5')
+      expect(modal.getByRole('heading', { name: 'charlie' }))
 
-      expect(args.batchTx).toHaveBeenCalledTimes(1)
+      await userEvent.click(getButtonByText(modal, 'Sign and Bond'))
+    })
+
+    await step('Add second validator account', async () => {
+      expect(await modal.findByText('You are intending to bond your validator account with your membership.'))
+      expect(modal.getByText('Transaction fee:')?.nextSibling?.textContent).toBe('5')
+      expect(modal.getByRole('heading', { name: 'dave' }))
+
+      await userEvent.click(getButtonByText(modal, 'Sign and Bond'))
+    })
+
+    await step('Confirm validator account', async () => {
+      expect(
+        await modal.findByText('You are intending to confirm your validator account to be bound with your membership')
+      )
+      expect(modal.getByText('Transaction fee:')?.nextSibling?.textContent).toBe('5')
+      expect(modal.getByRole('heading', { name: 'bob' }))
+
+      await userEvent.click(getButtonByText(modal, 'Sign and Confirm'))
+    })
+
+    await step('Confirm', async () => {
+      expect(await modal.findByText('Success'))
+      expect(modal.getByText(NEW_MEMBER_DATA.handle))
+      expect(args.onBuyMembership).toHaveBeenCalledWith({
+        rootAccount: alice.controllerAccount,
+        controllerAccount: bob.controllerAccount,
+        handle: NEW_MEMBER_DATA.handle,
+        metadata: metadataToBytes(MembershipMetadata, {
+          name: NEW_MEMBER_DATA.metadata.name,
+          about: NEW_MEMBER_DATA.metadata.about,
+          avatarUri: NEW_MEMBER_DATA.metadata.avatar.avatarUri,
+        }),
+        invitingMemberId: undefined,
+        referrerId: undefined,
+      })
+      expect(args.onAddStakingAccount).toHaveBeenCalledTimes(4) // means 2 times, due to the React hook 'useMomo'
```

Comment:
I don't think that's correct. `onSend` is only called when the mock transaction `signAndSend` method is called. So this should only happen once per transaction (so 2 times) regardless of the `useMemo`.

Am I missing something ? If not could you look into why this is called 4 times right now please ?

Reply:
Actually this extrinsic has called two times for each sign, first time is to get the fee of the tx, and next time for signature.

https://github.com/Joystream/pioneer/blob/5d255aed7ea1f8ffd2db41074995c43cef59ff37/packages/ui/src/common/hooks/useProcessTransaction.ts#L108
https://github.com/Joystream/pioneer/blob/5d255aed7ea1f8ffd2db41074995c43cef59ff37/packages/ui/src/common/hooks/useProcessTransaction.ts#L127

Reply:
Here's the mocked transaction code:

https://github.com/Joystream/pioneer/blob/4b9a450e210c0e14d005eca5fb7c75c94ea26594/packages/ui/src/mocks/helpers/transactions.ts#L30-L40

As you can see the `onSend` is only called in `signAndSend` not in `paymentInfo`. There must be a different reason why these transactions get sent multiple times.